sycl/test/native_cpu/math_builtins.cpp

```diff
@@ -0,0 +1,134 @@
+// REQUIRES: native_cpu_be
```

Comment:
What is specific about native CPU in this test? It seems like it could run on other targets too.

Also aren't there tests in the E2E suite already testing the math built-ins?

Reply:
I think this breaks the convention that runtime e2e tests are placed in sycl/test-e2e; whereas sycl/test checks device code generation etc. As @npmiller says ideally it is possible to use existing tests. This makes sure the test folders don't become cluttered with duplicate test cases. For example the `sycl::` functions you test are already covered in tests like https://github.com/intel/llvm-test-suite/blob/2e687fcd57149112d7b29e9e85bf74ce85f1119c/SYCL/DeviceLib/built-ins/scalar_math.cpp#L125

For the `sycl::native` function calls, I see it can make sense to test them on a per backend basis since the precision is generally backend dependent. Note FYI that they are already generally tested to some extent in this test: https://github.com/intel/llvm/blob/sycl/sycl/test-e2e/DeviceLib/built-ins/fast-math-flag.cpp

However the above fast-math-flag.cpp test is effectively just checking that the fast-math flag work such that `sycl::` math correctly call the `sycl::native` variants when the flag is used.

Reply:
Yeah you are right, I think that what you say applied to a lot of tests in `sycl/test/native_cpu` (e.g. we have a test for vector addition, I'm sure that equivalent functionalities are already tested in `e2e`). We will remove most of the tests in `sycl/test/native_cpu` once we enable e2e testing for it, but I've removed the test added in this PR, since we can reuse the tests that you pointed out.